libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -698,25 +880,25 @@ class MapboxRouteLineApi(
 
         val wayPointsFeatureCollection = wayPointsFeatureCollectionDef.await()
 
-        // The RouteLineExpressionData is only needed if the vanishing route line feature is
-        // enabled. The data is not needed for the rendering of the lines so there is no wait
-        // for completion.
-        if (routeLineOptions.vanishingRouteLine != null) {
-            jobControl.scope.launch(ThreadController.IODispatcher) {
-                val segments: List<RouteLineExpressionData> =
-                    partitionedRoutes.first.firstOrNull()?.route?.run {
-                        MapboxRouteLineUtils.calculateRouteLineSegments(
-                            this,
-                            routeLineOptions.resourceProvider.trafficBackfillRoadClasses,
-                            true,
-                            routeLineOptions.resourceProvider.routeLineColorResources,
-                            routeLineOptions.resourceProvider.restrictedRoadSectionScale,
-                            routeLineOptions.displayRestrictedRoadSections
-                        )
-                    } ?: listOf()
-                routeLineExpressionData.clear()
-                routeLineExpressionData.addAll(segments)
+        // The RouteLineExpressionData is only needed if the vanishing route line feature
+        // or styleInactiveRouteLegsIndependently feature are enabled.
+        if (
+            routeLineOptions.vanishingRouteLine != null ||
+            routeLineOptions.styleInactiveRouteLegsIndependently
+        ) {
+            val segmentsDef = jobControl.scope.async(ThreadController.IODispatcher) {
+                partitionedRoutes.first.firstOrNull()?.route?.run {
+                    MapboxRouteLineUtils.calculateRouteLineSegments(
+                        this,
+                        routeLineOptions.resourceProvider.trafficBackfillRoadClasses,
+                        true,
+                        routeLineOptions.resourceProvider.routeLineColorResources,
+                        routeLineOptions.resourceProvider.restrictedRoadSectionScale,
+                        routeLineOptions.displayRestrictedRoadSections
+                    )
+                } ?: listOf()
             }
+            routeLineExpressionData = segmentsDef.await()
```

Comment:
I believe we're regressing the time to first content here - previously, this code was executed completely independently of the callback delivered to the developer with first route geometries. Now, we're suspending the invocation of the callback until the expression data is parsed.

I did 3 tests using [this cross US route](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavui-maps/src/test/resources/cross-country-route.json) and on `main` the results of calling `MapboxRouteLineApi#setRoutes` were between 60-80ms, while now I've seen them between 150-250ms on my device, and I suspect this deferred is the reason.

If the old (detached) implementation worked correctly, we should leave it and only change the assignment point to be on the main thread to resolve the issues we discussed:
```diff
diff --git a/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt b/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt
index 5aff58ae1..942131078 100644
--- a/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt
+++ b/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt
@@ -886,19 +886,21 @@ class MapboxRouteLineApi(
             routeLineOptions.vanishingRouteLine != null ||
             routeLineOptions.styleInactiveRouteLegsIndependently
         ) {
-            val segmentsDef = jobControl.scope.async(ThreadController.IODispatcher) {
-                partitionedRoutes.first.firstOrNull()?.route?.run {
-                    MapboxRouteLineUtils.calculateRouteLineSegments(
-                        this,
-                        routeLineOptions.resourceProvider.trafficBackfillRoadClasses,
-                        true,
-                        routeLineOptions.resourceProvider.routeLineColorResources,
-                        routeLineOptions.resourceProvider.restrictedRoadSectionScale,
-                        routeLineOptions.displayRestrictedRoadSections
-                    )
-                } ?: listOf()
+            jobControl.scope.launch {
+                val segmentsDef = jobControl.scope.async(ThreadController.IODispatcher) {
+                    partitionedRoutes.first.firstOrNull()?.route?.run {
+                        MapboxRouteLineUtils.calculateRouteLineSegments(
+                            this,
+                            routeLineOptions.resourceProvider.trafficBackfillRoadClasses,
+                            true,
+                            routeLineOptions.resourceProvider.routeLineColorResources,
+                            routeLineOptions.resourceProvider.restrictedRoadSectionScale,
+                            routeLineOptions.displayRestrictedRoadSections
+                        )
+                    } ?: listOf()
+                }
+                routeLineExpressionData = segmentsDef.await()
             }
-            routeLineExpressionData = segmentsDef.await()
         }
 
         // This call is resource intensive so it needs to come last so that
```

Reply:
ok, i'll wrap it in a `launch` block.